### ISSUE TITLE
Add ability to reset all activities on a page

### ISSUE THREAD

**Issue by halturner** [Simon-Initiative/oli-torus#3469]
When the Submit function was removed from Multiple Choice questions, it appears the Reset function was also removed.

Parity with Legacy would require a Reset button. Students use it from time to time, but the most common use case is for demonstrations where an OLI rep will use the same pages and questions repeatedly. The inability to reset a question would make that impossible.
<img width="1124" alt="Screenshot 2023-05-03 at 10 33 15 AM" src="https://user-images.githubusercontent.com/24977416/235947820-41e6d5bd-0096-4434-82fb-88905b99c8d3.png">

**Comment by darrensiegel** [comment]
In Legacy, MCQs do not have a reset feature.  The Reset feature is at the level of the entire inline assessment.  There is no longer a notion of inline assessments in torus, which would make replicating this feature difficult.  Is this really a needed feature?

**Comment by czerwinskie** [comment]
Yes.  Students use these formative assessments to practice and study for an exam, so resetting them is necessary for them to be used in that way.

**Comment by darrensiegel** [comment]
So as it stands right now, every activity type within Torus in a practice page can be attempted an unlimited number of times.  

Are we saying that we also need a single "Reset" button that resets all activities on a page back to their empty state?

**Comment by czerwinskie** [comment]
Yes, if students are trying to re-study for an exam, it doesn't help if they can see their previous responses. There is an underlying philosophy to the approach we took in legacy that maybe Norm can articulate to help us determine what we do here.  I think in legacy, there was even a way to reset all, but not sure about that.

**Comment by normanbier** [comment]
- It may be desirable to reset all activities on a page at once, but it is not a feature that exists w/I legacy OLI.  This would definitely be a new capability and is not one I would prioritize for v24
- The ability to reset an entire activity -- ie all of the questions in an activity at once -- is something that is currently in legal and I do think is important -- students do use this to reattempt activities, study, etc.
- Needing to individually reset each question in an activity (multiple buttons per activity) is highly undesirable. The UX element of it (multiple interactions) isn't great.  And pedagogically, if I have a multi-step set of related questions, needing to change the reset state as I work through the question as a student isn't ideal.
- This capability is most important for questions are templated and/or free response (entering a number, text, etc.).  It is less immediately critical for fixed response (MC and drop-down). If we needed to we could live with the missing reset on MC and drop-down.  We did not have any reset capabilities in Legacy for many years, this is something that was added in like '15 or '16, and when it was added it applied to all types of questions (I think it was managed via a theme fix, though I don't remember).

So I believe this should get split out into multiple tickets:

1. Parity need: single button reset of all questions in an activity (higher priority)
2. Parity need: reset option for fixed choice questions (lower priority, may be subsumed into #1, above)
3. Feature request: Mass reset ability. This needs more time and research to flesh out.  Is it needed at all? Should it be full page reset or if the goal is simply for exam prep, wouldn't a module reset be a better experience? Not MVP, for sure.

Last point: it's becoming clearer to me that Torus is less opinionated about what constitutes an activity than is Legacy; this is creating some disconnects in expectations esp. in LE/ID/UX/DS approaches.  We should talk more about this and what it means short and long term, maybe in the upcoming migration or feature planning meeting.
